app/src/main/jni/audiolat.cpp

```diff
@@ -42,36 +43,38 @@ aaudio_data_callback_result_t dataCallback(AAudioStream *stream, void *userData,
   if (stream == cb_data->record_stream) {
     LOGD("record num_frames: %d time_sec: %.2f", num_frames, time_sec);
     // recording
-    if (time_sec - last_ts > 2) {
+    if (time_sec - last_ts > 2 || rec_buffer_index > 0) {
       playing = true;
-      LOGD("record write: input num_frames: %d", num_frames - cb_data->begin_signal_size);
+      LOGD("record write begin signal: input num_frames: %d", num_frames - cb_data->begin_signal_size);
```

Comment:
this log should be:

    "record write: begin num_frames: %d"

Reply:
also, please follow the linux convention for patch names:


   audiolat: support being signals spanning several buffers

Reply:
> also, please follow the linux convention for patch names:
> 
> audiolat: support being signals spanning several buffers

syntax: <subsystem>: <subject_line>

where:
* subsystem: "audiolat" in this case
* subject_line: text that makes sense in the sentence “If applied, this commit will <subject_line>”